### PR TITLE
Messenger 21

### PR DESCRIPTION
**Pull request by Nikita8Sannikov**
https://tracker.yandex.ru/MESSANGER-21
- [x] add Link component
- [x]  replace Link tags with our components

### CODE REVIEW

File: app/src/components/Common/Link/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Link as MuiLink, LinkProps } from "@mui/material";
+
+type Ilink = {
```
Comment (makebelieve-sk) [inline app/src/components/Common/Link/index.tsx:3]: why you choose this name of the interface?
type - ExampleType
interface - IExample

File: app/src/modules/Profile/Photos/index.tsx [hunk 2]
```diff
@@ -13,90 +13,108 @@ import Photo from "../../../components/Common/Photo";
...
+						variant="body2"
+						onClick={() => navigate(Pages.photos)}
+						underline="hover"
+						className="photo-container__title__link"
+					>
+						Посмотреть все
+					</LinkComponent>
+				</div>
+
+				{loadingPhotos ? (
```
Comment (makebelieve-sk) [inline app/src/modules/Profile/Photos/index.tsx:88]: ? and : on the next line

File: app/src/modules/Profile/Photos/index.tsx [hunk 3]
```diff
@@ -13,90 +13,108 @@ import Photo from "../../../components/Common/Photo";
...
+					id="add-new-photos"
+					text="Добавить еще"
+					loading={loadingBtn}
+					multiple
+					onChange={addPhotosHandler}
+				/>
+			</Paper>
+		</Grid>
+	);
+});
```
Comment (makebelieve-sk) [inline app/src/modules/Profile/Photos/index.tsx:120]: also, you forgot the auth pages. On the SignIn and SignUp pages you have several links. For example, Забыли пароль? or Нет аккаунта? Зарегистрируйтесь! (now, we havent this page, but its coming soon), Уже есть аккаунт? Войдите

File: app/src/components/Common/Link/index.tsx (hunk 1 above)
Comment (Nikita8Sannikov) [reply app/src/components/Common/Link/index.tsx:3]: oh i got it? we need to name it LinkType
right?

File: app/src/modules/Profile/Photos/index.tsx (hunk 2 above)
Comment (Nikita8Sannikov) [reply app/src/modules/Profile/Photos/index.tsx:88]: my prettier automatically fix this way when i save the file
tell me please your setting

File: app/src/modules/Profile/Photos/index.tsx (hunk 3 above)
Comment (Nikita8Sannikov) [reply app/src/modules/Profile/Photos/index.tsx:120]: Aren't they made with the help of "navigate" to redirect to another Route ?
and the task was replace all uses of <a> or <Link>

File: app/src/components/Common/Link/index.tsx (hunk 1 above)
Comment (makebelieve-sk) [reply app/src/components/Common/Link/index.tsx:3]: yes!

Comment (makebelieve-sk) [reply app/src/components/Common/Link/index.tsx:3]: may be LinkComponentType would be better?

File: app/src/components/Common/Link/index.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	children?: React.ReactNode;
+} & LinkProps;
+
+export default function LinkComponent({
+	children,
+	onClick,
+	href,
+	...props
+}: LinkComponentType) {
+	const onEvent = useCallback(
```
Comment (makebelieve-sk) [inline app/src/components/Common/Link/index.tsx:16]: op-pa
nashla koca na kamenb
for what purposes do we use useCallback? What are the differences between useCallback and useMemo? When should useCallback and useMemo be used and when should they be ignored? I think we should discuss this in telegramm

File: app/src/pages/SignUp.tsx [hunk 5]
```diff
@@ -1,272 +1,285 @@
...
-import Box from "@mui/material/Box"
-import LockOutlinedIcon from "@mui/icons-material/LockOutlined"
-import Typography from "@mui/material/Typography"
-import Container from "@mui/material/Container"
-import { createTheme, ThemeProvider } from "@mui/material/styles"
-import Stepper from "@mui/material/Stepper"
-import Step from "@mui/material/Step"
-import StepLabel from "@mui/material/StepLabel"
-import Button from "@mui/material/Button"
+import * as React from "react";
```
Comment (makebelieve-sk) [inline app/src/pages/SignUp.tsx:1]: Remind me please, why we use import * as React from "react", but not use import React from "react"?

File: app/src/components/Copyright/index.tsx [hunk 6]
```diff
@@ -1,21 +1,25 @@
...
-        </div>
-    );
-});
\ No newline at end of file
+	return (
+		<div className="copyright">
+			Copyright ©
+			<LinkComponent
+				href={Pages.aboutUs}
+				variant="body2"
```
Comment (makebelieve-sk) [inline app/src/components/Copyright/index.tsx:17]: Also, i see that in all places of use the prop variant="body2" is passed. Maybe we can use this prop inside the LinkComponent?

File: app/src/pages/SignUp.tsx (hunk 5 above)
Comment (Nikita8Sannikov) [reply app/src/pages/SignUp.tsx:1]: I suggested to import this way like:
import { useEffect, useState } from 'react'

but i haven't fix it in this task

File: app/src/components/Copyright/index.tsx (hunk 6 above)
Comment (Nikita8Sannikov) [reply app/src/components/Copyright/index.tsx:17]: but if one day we will not use this prop?
i thought the component should be more universal

File: app/src/components/Common/Link/index.tsx (hunk 4 above)
Comment (Nikita8Sannikov) [reply app/src/components/Common/Link/index.tsx:16]: oopsii, let's talk abou it

File: app/src/components/Copyright/index.tsx (hunk 6 above)
Comment (makebelieve-sk) [reply app/src/components/Copyright/index.tsx:17]: read please about YAGNI principle
after that you can change your opinion)

File: app/src/pages/SignIn.tsx [hunk 7]
```diff
@@ -1,194 +1,217 @@
...
+							...formValues.errors,
+							login: error.message,
+							password: error.message,
+						},
+					});
+				}
+			});
+		}
+	};
+	console.log("render");
```
Comment (makebelieve-sk) [inline app/src/pages/SignIn.tsx:90]: delete this please, may be i forgot that shit

File: app/src/pages/SignUp.tsx (hunk 5 above)
Comment (makebelieve-sk) [reply app/src/pages/SignUp.tsx:1]: of cource, it not a problem. 
Okey, maybe we will come back to this question later

**Pull request merged**